pkg/db/client.go

```diff
@@ -238,6 +255,30 @@ func (t *TridentDB) ListCampaign() ([]Campaign, error) {
 	return campaigns, nil
 }
 
+// IsCampaignCancelled takes a campaign ID and returns true if the campaign status is CampaignStatusCancelled
+func (t *TridentDB) IsCampaignCancelled(campaignID uint) (bool, error) {
+	var matches []Campaign
+
+	var query = Query{
+		Filter: map[string]interface{}{"id": campaignID, "status": CampaignStatusCancelled},
+	}
+
+	err := t.db.
+		Where(query.Filter).
+		Find(&matches).
+		Error
+
+	if err != nil {
+		return false, err
+	}
+
+	if len(matches) != 0 {
```

Comment:
Can we do this at the database level to avoid the performance hit of actually querying all this data? Campaigns contain a LOT of data since they have the usernames and passwords.

e.g. Add a `.Count(&count)` at the end and check the count. https://gorm.io/docs/advanced_query.html#Count

Reply:
We can also simply the logic for the return here to something like
```
return count > 0, nil
```